jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataContext.java

```diff
@@ -336,6 +336,13 @@ public DataSet executeQuery(CompiledQuery compiledQuery, Object... values) {
 
     private DataSet execute(Connection connection, Query query, Statement statement, JdbcCompiledQuery compiledQuery,
             JdbcCompiledQueryLease lease, Object[] values) throws SQLException, MetaModelException {
+        Integer maxRows = query.getMaxRows();
+
+        final List<SelectItem> selectItems = query.getSelectClause().getItems();
+        if (maxRows != null && maxRows.intValue() == 0) {
```

Comment:
Then the next guy doesn't try to understand the code he is changing, and shouldn't be allowed near it ;)